python/larpix.py

```diff
@@ -16,43 +16,155 @@ def __init__(self, chip_id, io_chain):
-        self._global_threshold = 0
-        self._csa_gain = 0
-        self._csa_bypass = 0
-        self._internal_bypass = 0
-        self._csa_bypass_select = [0] * 32
-        self._csa_monitor_select = [0] * 32
-        self._csa_testpulse_enable = [0] * 32
-        self._csa_testpulse_dac_amplitude = [0] * 32
-        self._test_mode = 0
-        self._cross_trigger_mode = 0
-        self._periodic_reset = 0
-        self._fifo_diagnostic = 0
-        self._sample_cycles = 0
-        self._test_burst_length = 0
-        self._adc_burst_length = 0
-        self._channel_mask = 0
-        self._external_trigger_mask = 0
-        self._reset_cycles = 0
+        self.pixel_trim_thresholds = [0x10] * larpix.larpix_num_channels()
+        self.global_threshold = 0x10
+        self.csa_gain = 1
+        self.csa_bypass = 0
+        self.internal_bypass = 1
+        self.csa_bypass_select = [0] * larpix.larpix_num_channels()
+        self.csa_monitor_select = [1] * larpix.larpix_num_channels()
+        self.csa_testpulse_enable = [0] * larpix.larpix_num_channels()
+        self.csa_testpulse_dac_amplitude = 0
+        self.test_mode = 0
+        self.cross_trigger_mode = 0
+        self.periodic_reset = 0
+        self.fifo_diagnostic = 0
+        self.sample_cycles = 1
+        self.test_burst_length = 0x00FF
+        self.adc_burst_length = 0
+        self.channel_mask = 0
+        self.external_trigger_mask = 0xFFFFFFF
+        self.reset_cycles = 0x001000
+        
+    def get_ctypes_configuration(self):
+        ctypes_config = l.larpix_configuration()
+
+        for i,value in enumerate(self._pixel_trim_thresholds):
+            ctypes_config.pixel_trim_thresholds[i] = value
+        ctypes_config.global_threshold = self._global_threshold
+        ctypes_config.csa_gain = self._csa_gain
+        ctypes_config.csa_bypass = self._csa_bypass
+        ctypes_config.internal_bypass = self._internal_bypass
+        for i,value in enumerate(self._csa_bypass_select):
+            ctypes_config.csa_bypass_select[i] = value
+        for i,value in enumerate(self._csa_monitor_select):
+            ctypes_config.csa_monitor_select[i] = value
+        for i,value in enumerate(self._csa_testpulse_enable):
+            ctypes_config.csa_testpulse_enable[i] = value
+        ctypes_config.csa_testpulse_dac_amplitude = self._csa_testpulse_dac_amplitude
+        ctypes_config.test_mode = self._test_mode
+        ctypes_config.cross_trigger_mode = self._cross_trigger_mode
+        ctypes_config.periodic_reset = self._periodic_reset
+        ctypes_config.fifo_diagnostic = self._fifo_diagnostic
+        ctypes_config.sample_cycles = self._sample_cycles
+        for i,value in enumerate(self._test_burst_length):
+            ctypes_config.test_burst_length[i] = value
+        ctypes_config.adc_burst_length = self._adc_burst_length
+        for i,value in enumerate(self._channel_mask):
+            ctypes_config.channel_mask[i] = value
```

Comment:
Here you treat `self._channel_mask` as a list but you define it as a bitmap at Line 132. This is what got me mixed up.